phoebe/backend/backends.py

```diff
@@ -1151,16 +1161,47 @@ def _run_single_time(self, b, i, time, infolist, **kwargs):
             logger.debug("rank:{}/{} PhoebeBackend._run_single_time: calling system.populate_observables at time={}".format(mpi.myrank, mpi.nprocs, time))
             system.populate_observables(time, populate_kinds, populate_datasets)
 
+        # save temporary positions (for interferometry)
+        system.xi = xi
+        system.yi = yi
+        system.zi = zi
+        system.distance = b.get_value('distance@system')
+
         logger.debug("rank:{}/{} PhoebeBackend._run_single_time: filling packets at time={}".format(mpi.myrank, mpi.nprocs, time))
         # now let's loop through and prepare a packet which will fill the synthetics
         packetlist = []
+        previous = None
         for k, info in enumerate(infolist):
             packet = dict()
 
-            # i, time, info['kind'], info['component'], info['dataset']
+            #i, time, info['kind'], info['component'], info['dataset']
             cind = starrefs.index(info['component']) if info['component'] in starrefs else None
-            # ts[i], xs[cind][i], ys[cind][i], zs[cind][i], vxs[cind][i], vys[cind][i], vzs[cind][i]
+            #ts[i], xs[cind][i], ys[cind][i], zs[cind][i], vxs[cind][i], vys[cind][i], vzs[cind][i]
             kind = info['kind']
+            dataset = info['dataset']
+
+            # save baselines and wavelengths (for interferometry)
+            if kind == 'vis' and dataset != previous:
```

Comment:
is the need for `previous` because of the "flattening" and using `original_index`?  Is there any concern if the input is not sorted correctly?

Reply:
Here, it is useless to save baselines and wavelengths in each cycle (over k, info). It is an useful optimization. Btw. the b.get_value() calls are pretty expensive, especially if present in the innermost cycle!

Reply:
Yes, `get_value` calls should be minimized as much as possible (by pulling any values in advance that need to be accessed multiple times, etc).  Since you're pulling multiple values from the dataset, you can also save quite a bit of time by skipping multiple visibility checks and by pre-filtering to a smaller common set of parameters so that future calls have less parameters to search through.  For example:

```
ds_ps = b.filter(dataset=dataset, context='dataset', **_skip_filter_checks)
...
ucoord = ds_ps.get_value(qualifier='u', **_skip_filter_checks)
vcoord = ds_ps.get_value(qualifier='v', **_skip_filter_checks)
```